src/main/java/ch/njol/skript/expressions/ExprMaxStack.java

```diff
@@ -18,35 +18,131 @@
+@Description({
+	"The maximum stack size of an item (e.g. 64 for torches, 16 for buckets, and 1 for swords), or of an inventory to have items to stack up to.",
+	"Since MC 1.20.5 onwards, the maximum stack size of items can be changed (any integer from 1 to 99), and the said item can be stacked up to the set value, up to the maximum stack size of an inventory."
+})
+@Examples({
+	"send \"You can only pick up %max stack size of player's tool% of %type of (player's tool)%\" to player",
+	"",
+	"set the maximum stack size of inventory of all players to 16",
+	"",
+	"add 8 to the maximum stack size of player's tool",
+	"",
+	"reset the maximum stack size of {_gui}"
+})
+@Since("2.1, INSERT VERSION (changeable, inventory support)")
+@RequiredPlugins("Spigot 1.20.5+ (changeable item max stack size)")
+public class ExprMaxStack extends SimplePropertyExpression<Object, Integer> {
 
 	static {
-		register(ExprMaxStack.class, Long.class, "max[imum] stack[[ ]size]", "itemtype");
+		register(ExprMaxStack.class, Integer.class, "max[imum] stack[[ ]size]", "itemtypes/inventories");
+	}
+
+	private static final boolean CHANGEABLE_ITEM_STACK_SIZE = Skript.methodExists(ItemMeta.class, "setMaxStackSize", Integer.class);
+
+	@Override
+	@Nullable
+	public Integer convert(Object source) {
+		if (source instanceof ItemType) {
+			return ((ItemType) source).getRandom().getMaxStackSize();
+		} else if (source instanceof Inventory) {
+			return (((Inventory) source).getMaxStackSize());
+		} else {
+			// Invalid source -- return null
+			return null;
+		}
 	}
-	
-	@SuppressWarnings("null")
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+            case ADD:
+			case DELETE:
+            case REMOVE:
+			case RESET:
+			case SET:
+				if (!CHANGEABLE_ITEM_STACK_SIZE && ItemType.class.isAssignableFrom(getExpr().getReturnType())) {
+					Skript.error("Changers for maximum stack size of items requires Minecraft 1.20.5 or newer");
+					return null;
+				}
+				return CollectionUtils.array(Integer.class);
+			default:
+				return null;
+        }
+	}
+
 	@Override
-	public Long convert(final ItemType i) {
-		return (long) i.getRandom().getMaxStackSize();
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		for (Object source : getExpr().getArray(event)) {
+			Integer change = null;
+			if (mode != ChangeMode.DELETE && mode != ChangeMode.RESET)
+				change = (int) delta[0];
+			if (source instanceof ItemType) {
+				if (!CHANGEABLE_ITEM_STACK_SIZE)
+					continue;
+				ItemType itemType = ((ItemType) source);
+				int size = itemType.getRandom().getMaxStackSize();
+                switch (mode) {
+                    case ADD:
+						size += change;
+                        break;
+                    case SET:
+						size = change;
+						break;
+                    case REMOVE:
+						size -= change;
+                        break;
```

Comment:
Reset and Delete are not handled.

Reply:
Here where it handles:
https://github.com/SkriptLang/Skript/blob/9e5833800a5d37ff70d7b2413de0bb40a32caaa6/src/main/java/ch/njol/skript/expressions/ExprMaxStack.java#L98-L99

Then here:
https://github.com/SkriptLang/Skript/blob/9e5833800a5d37ff70d7b2413de0bb40a32caaa6/src/main/java/ch/njol/skript/expressions/ExprMaxStack.java#L118

Reply:
Ah I see, I missed the ternary